plugins/admin/pkg/command/registry/remove.go

```diff
@@ -0,0 +1,148 @@
+				LabelSelector: labels.SelectorFromSet(AdminRegistryLabels).String(),
+			})
+			if err != nil {
+				return fmt.Errorf("failed to list secret: %v", err)
+			}
+
+			// filter the secrets with username and server
+			secretsMap := make(map[string]*corev1.Secret)
+			for _, secret := range secrets.Items {
+				registry := Registry{}
+				err = json.Unmarshal(secret.Data[DockerJSONName], &registry)
+				if err != nil {
+					return fmt.Errorf("failed unmarshal secret data '.dockerconfigjson': %v", err)
+				}
+				for secretServer, secretAuth := range registry.Auths {
+					if secretServer == server && secretAuth.Username == username {
+						secretsMap[secret.Name] = &secret
+					}
+				}
+			}
+			if len(secretsMap) == 0 {
+				cmd.Printf("No registry found for server: '%s' and username: '%s'\n", server, username)
+				return nil
+			}
+
+			defaultSa, err := p.ClientSet.CoreV1().ServiceAccounts("default").Get("default", metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get serviceaccount: %v", err)
+			}
+
+			desiredSa := defaultSa.DeepCopy()
+			imagePullSecrets := []corev1.LocalObjectReference{}
+			for _, ips := range desiredSa.ImagePullSecrets {
+				if _, ok := secretsMap[ips.Name]; !ok {
+					// only store the secrets that do not exist in the map
+					imagePullSecrets = append(imagePullSecrets, ips)
+				}
+
+			}
+
+			desiredSa.ImagePullSecrets = imagePullSecrets
+			_, err = p.ClientSet.CoreV1().ServiceAccounts("default").Update(desiredSa)
+			if err != nil {
+				return fmt.Errorf("failed to remove registry secret in default serviceaccount: %v", err)
+			}
+			cmd.Printf("ImagePullSecrets of ServiceAccount '%s/%s' updated\n", desiredSa.Namespace, desiredSa.Name)
+
+			deleteSecretsErrCh := make(chan error, len(secretsMap))
+			deleteSecrets(cmd, p.ClientSet, secretsMap, deleteSecretsErrCh)
+
+			var deleteSecretsErr error
+			select {
+			case deleteSecretsErr = <-deleteSecretsErrCh:
+				if deleteSecretsErr != nil {
+					break
+				}
+			default:
+			}
+
+			if deleteSecretsErr != nil {
+				return fmt.Errorf("failed to delete secrets: %v", deleteSecretsErr)
+			}
+
+			return nil
+		},
+	}
+
+	prRmCmd.Flags().StringVar(&username, "username", "", "Registry Username")
+	prRmCmd.MarkFlagRequired("username")
+	prRmCmd.Flags().StringVar(&server, "server", "", "Registry Address")
+	prRmCmd.MarkFlagRequired("server")
+	prRmCmd.InitDefaultHelpFlag()
+	return prRmCmd
+}
+
+func deleteSecrets(cmd *cobra.Command, clientset kubernetes.Interface, secretsMap map[string]*corev1.Secret, errCh chan<- error) {
+	w := sync.WaitGroup{}
+	w.Add(len(secretsMap))
+	for _, s := range secretsMap {
+		go func(secret *corev1.Secret) {
+			defer w.Done()
+			err := clientset.CoreV1().Secrets(secret.Namespace).Delete(secret.Name, &metav1.DeleteOptions{})
+			if err != nil {
+				errCh <- fmt.Errorf("failed to delete secret '%s': %v", secret.Name, err)
```

Comment:
If secret delete error is something like `apierrors.IsNotFound(err)`, this is also what we expected as we want to delete it.
I'm thinking we do not need to return error here, what do you think?

Reply:
Fixed. The `not found` message will be printed out instead of throwing an error.